### PR TITLE
Add export URLs and content types to published models endpoints and Atom feed

### PR DESCRIPTION
**Pull request by joe-crawford**
Changes for the 'publish' side of federation using MIME content types (part of mc-9618):

- Add XML versions of Mauro published model endpoints
- Add XSDs (in mdm-plugin-federation/grails-app/assets/xsd) for Atom published model feed, and Mauro published model XML endpoints and validate in tests.
- Add exporter links and content type to published model and Atom responses (mc-9831, mc-9832)
- Add content type string to ExporterProviderServices (part of mc-9835)

### CODE REVIEW

**Comment by olliefreeman** [comment]
dont worry about retrying this, the tests all passed, it got stuck on the SCA so i stopped the job

**Review by olliefreeman — CHANGES_REQUESTED** [review]
A few small changes, otherwise looks good

File: mdm-core/src/main/groovy/uk/ac/ox/softeng/maurodatamapper/core/provider/exporter/ExporterProviderService.groovy [hunk 1]
```diff
@@ -43,8 +43,36 @@ abstract class ExporterProviderService extends MauroDataMapperService {
...
+     */
+    String getProducesContentType() {
+        null
+    }
+
+    /**
+     * Flag preferred/native ExporterProviderService, to be sorted first.
+     * @return true if preferred
+     */
+    boolean getIsPreferred() {
```
[the remark this answers is not in the document]
Comment (joe-crawford) [reply mdm-core/src/main/groovy/uk/ac/ox/softeng/maurodatamapper/core/provider/exporter/ExporterProviderService.groovy]: Have removed. Intention was to separate out the native exporters so they would always be rendered first as Atom doesn't support any other way of indicating which link is preferred. I've set the order of the JSON exporters to HIGHEST_PRECEDENCE and the XML exporters to HIGHEST_PRECEDENCE+1.

File: mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/publish/PublishService.groovy [hunk 2]
```diff
@@ -23,60 +23,79 @@ import uk.ac.ox.softeng.maurodatamapper.core.model.Model
...
     }
 
     List<PublishedModel> findAllPublishedReadableModels(UserSecurityPolicyManager userSecurityPolicyManager) {
-        findAllReadableModelsToPublish(userSecurityPolicyManager).collect {new PublishedModel(it)}.sort()
+        List<PublishedModel> publishedModels = []
+        modelServices.each {modelService ->
+            List<Model> readableModels = modelService.findAllReadableModels(userSecurityPolicyManager, false, true, false)
+            // Only publish finalised models which belong to this instance of MDM
+            List<Model> publishableModels = readableModels.findAll {Model model ->
+                model.finalised && model.authority.id == authorityService.getDefaultAuthority().id
```
[the remark this answers is not in the document]
Comment (joe-crawford) [reply mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/publish/PublishService.groovy]: Resolved

File: mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/publish/PublishService.groovy [hunk 3]
```diff
@@ -23,60 +23,79 @@ import uk.ac.ox.softeng.maurodatamapper.core.model.Model
...
-            // Only publish finalised models which belong to this instance of MDM
-            List<Model> publishableModels = readableModels.findAll {Model model ->
-                model.finalised && model.authority.id == authorityService.getDefaultAuthority().id
-            } as List<Model>
-            models.addAll(publishableModels)
+    PublishedModel getPublishedModel(Model model, ModelService modelService) {
+        String modelUrl = linkGenerator.link(resource: model, method: HttpMethod.GET, absolute: true)
+        new PublishedModel(model).tap {
+            links = modelService.getExporterProviderServices().sort().collect {exporterProviderService ->
+                new Link(MdmAtomPublishedModelRenderer.RELATIONSHIP_ALTERNATE,
```
[the remark this answers is not in the document]
Comment (joe-crawford) [reply mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/publish/PublishService.groovy]: Resolved

File: mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/SubscribedCatalogueService.groovy [hunk 4]
```diff
@@ -138,6 +140,7 @@ class SubscribedCatalogueService implements XmlImportMapping, AnonymisableServic
                 datePublished = OffsetDateTime.parse(pm.datePublished, DateTimeFormatter.ISO_OFFSET_DATE_TIME)
                 author = pm.author
                 description = pm.description
+                if (pm.links) links = pm.links.collect {link -> new Link(MdmAtomPublishedModelRenderer.RELATIONSHIP_ALTERNATE, link.url).tap {contentType = link.contentType}}
```
[the remark this answers is not in the document]
Comment (joe-crawford) [reply mdm-plugin-federation/grails-app/services/uk/ac/ox/softeng/maurodatamapper/federation/SubscribedCatalogueService.groovy]: Resolved

**Comment by olliefreeman** [comment]
test failures not due to this PR

**Pull request merged**